README.md

```diff
@@ -61,7 +61,7 @@ $ git secret reveal -f
 Developing with Livepush
 ------------------------
 
-To start developing, you must first install [balenaCLI](https://github.com/balena-io/balena-cli) and set up a local-mode device.
+To start developing, you must first install [balenaCLI](https://github.com/balena-io/balena-cli) and set up a local-mode device. For development machine, we recommend an Intel-NUC with i7 processor and at least 8 GB memory. You could create a virtual machine on VirtualBox following [this guide](https://www.balena.io/blog/no-hardware-use-virtualbox/), 32 GB of disk space, 8 GB RAM and 4 CPU cores are recommended. However, note that developing with a virtual machine, your machine might get overwhelmed by some tasks, like the initial push.
```

Comment:
We try and stay away from being too specific about the specific machine (NUC) as balena supports a large range of devices and JF should run on anything x86_64 that has enough resources.

Reply:
I honestly think that it would be useful to mention NUC explicitly. On the supported single board computers list, only another realistic physical board looks to me like UP Board. Others are emulated or generic. Although the idea is to support as many devices as possible, it looks to me that this is a reality for the current moment. And writing this on the documentation would make onboarding easier.

Reply:
[Single-board computers](https://www.balena.io/docs/reference/hardware/devices/)

Reply:
I tend to agree with @taromurao. Although in theory we support other devices, it seems that the vast majority of JF developers is using  the NUC + livepush. I faced some difficulties during my onboarding precisely because I tried to setup the development environment in my own laptop using `docker-compose` and `nodemon`. Once I understood that everyone else was using a NUC, and made the switch, everything became easier.

Reply:
@taromurao @edufschmidt We've had this conversation before and decided to stay away from being overly specific. I think we can mention the NUC as an example, but not go so far as to say that we "recommend" it. We want this to be accessible to the open-source community and making people think they need to spend hundreds of dollars on a NUC before they can work on it will drive most people away immediately.

Reply:
@edufschmidt I find it interesting that you tried to use `docker-compose` and `nodemon` before Livepush with a balena device since the README plainly states near the top `To start developing, you must first install balenaCLI and set up a local-mode device.`. Maybe we should update some of the links at the top to something like this:
```
...
- Developing locally (not recommended)
...
- Running in compose (not recommended for development)
...
```

Apparently we need more to direct people towards the recommended path of Livepush + a balena device.

Reply:
@joshbwlng @edufschmidt  I also tried docker-compose path too because when I got stuck with local-mode device with a VirtualBox VM, but now with the recent performance improvement of ui compilation at development https://github.com/product-os/jellyfish/pull/7714, development with VirtualBox VM works also fine. I have not yet found VM/container solutions for development. As far as I tried, deploying balenaOS on a Qemu is not easy by just following instructions on https://www.balena.io/docs/learn/getting-started/qemux86-64/python/, due to drive size problem and network configuration. Being able to running balenaOS directly on KVM would be also very useful, but I have had no luck yet.